app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageActivity.kt

```diff
@@ -71,6 +71,34 @@ class ZimManageActivity : BaseActivity() {
       offscreenPageLimit = sectionsPagerAdapter.count - 1
       tabs.setupWithViewPager(this)
       addOnPageChangeListener(SimplePageChangeListener(::updateMenu))
```

Comment:
`sectionsPagerAdapter.getItem(position)` will give you the fragment, use SimplePageChangeListener but just add another one

Reply:
@macgills I tried this earlier. It didn't work. What am I doing wrong?

```kotlin
override fun onPageScrolled(
          position: Int,
          positionOffset: Float,
          positionOffsetPixels: Int
        ) {
          if (position == 1) {
            val page = sectionsPagerAdapter.getItem(0)
            if (page != null) {
              (page as ZimFileSelectFragment).finishActionMode()
            }
          }
        }
```

> SimplePageChangeListener but just add another one

This means I should add the condition to the existing SimplePageChangeListener and add a new one in the ZimManageActivity.

Reply:
Use a debugger and find out

Reply:
>This means I should add the condition to the existing SimplePageChangeListener and add a new one in the ZimManageActivity.

`addOnPageChangeListener(SimplePageChangeListener(::updateActionMode))`

Reply:
@macgills I checked the values returned by `getItem()` and the method I have used. The values differ. I don't think that `getItem()` returns the exact value of the fragment that is needed to obtain a fragment instance.

Reply:
@macgills sorry, I am not very clear about how to send a signal to the LibraryFragment.
Do I have to communicate through the `ZimManageViewModel` and use `instantiateItem`?

Also could you please tell me what is wrong with the approach I have used currently?

Reply:
You are depending on internal apis to not change, a bad practice

```
    private static String makeFragmentName(int viewId, long id) {
        return "android:switcher:" + viewId + ":" + id;
    }
```

The androidx team have made no guarantee that this Id will stay the same (sidenote: it will likely never change because a billion apps rely on this fix so now we have hamstrung the androidx team so they cannot change. It is a shitty situation from all sides and I want to sidestep it, _don't depend on private methods_) and neither should we.

Create a publishProcessor in the viewmodel, maybe something like currentTabIndex, send it the currentPage when the page changes, in ZimFileSelectFragment subscribe to this processor. There are additional steps you can take to make this cleaner but this will do for the broad outline

Reply:
@macgills thanks a lot for this explanation. This made things pretty clear :+1: